pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -181,6 +199,36 @@ func (o *options) Run(ctx context.Context) error {
 		})
 	}
 
+	if o.version != "" {
+		target, err := semver.Parse(o.version)
+		if err != nil {
+			return fmt.Errorf("cannot parse SemVer target %q: %v", o.version, err)
+		}
```

Comment:
parse and check is redundant with the one done in `Complete` - could `Complete` save the parsed version to a another private `option` field and we'd use the parse thing here?

Reply:
I guess?  This function isn't a hot loop though, and I'm not sure the bookkeeping is worth it to save a few dozen re-parses.

Reply:
it's just less code to read and maintain, and `Complete` processing input into something used later is a common pattern :shrug: 

but yeah, no big deal, feel free to ignore